### PR TITLE
2143 save as visited 2

### LINKED ISSUE #2143: Programs: ensure user will update the encounter status
## Is your feature request related to a problem? Please describe 👀

When creating a new encounter, it is easy to miss the step when user is supposed to update the encounter's status cause it is located in the headers. 

The risk is that we will have a lot of encounter with default status "Pending" or "Missed" even though they are in fact "Visited" but user simply forgot to update the status. 

## Describe the solution you'd like 🎁

if encounter's status = `Pending` and new data has been added, when clicking on the `Save` button a popup window would ask user whether to update the encounter's status to "visited" or not 

### Describe alternatives you've considered 💭

We could also add another button in the bottom right corner entitled `Confirm XXX` (similar to what has been done for shipments' status update with the possibility to update the encounter's status to either `Visited` or `Cancelled`

Screenshot 2023-08-28 at 3 12 25 PM

We would need to keep `Save`  as a separate button since you could theoretically update an encounter without changing the status and user can still update an encounter even when status = "Visited".  

### Additional context 💌

Screenshot 2023-08-28 at 3 09 35 PM

### Moneyworks Jobcode 🧰

PG23THRFD

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes #2143 

The following approach has been chosen: The UI stays mostly as before, only if the encounter is in the past and the status Pending a change has been introduced.
In this case the save button shows a confirmation menu if the encounter shouldn't be saved as "Visited":

image

Happy to change the Modal design...

# Testing

- Create an encounter in the future: save button should never trigger the new dialog
- Create an encounter in the past with status Pending: dialog should show when saving
- Create and encounter in the past with status != Pending: dialog shouldn't show

### CODE REVIEW

File: client/packages/programs/src/schema_types.ts [hunk 1]
```diff
@@ -1,7 +1,21 @@
+import { EncounterNodeStatus } from '@common/types';
+import { Clinician } from 'packages/system/src/Clinician';
```
Comment (mark-prins) [inline client/packages/programs/src/schema_types.ts:2]: causes issues with tests (yes, I've explored using eslint to pick this up.. no success yet)

I've added this to `packages/system/src/index.ts`
```
export * from './Clinician';
```

and updated the import to 
```suggestion
import { Clinician } from '@openmsupply-client/system';
```

File: client/packages/system/src/Encounter/DetailView/DetailView.tsx [hunk 2]
```diff
@@ -43,6 +48,88 @@ const getPatientBreadcrumbSuffix = (
...
+    onSave,
+    encounterData,
+    updateEncounter
+  );
+
+  const SaveAsVisitedModal = () => (
+    <Modal
+      title={t('messages.save-encounter-as-visited')}
+      cancelButton={<DialogButton variant="cancel" onClick={hideDialog} />}
+      height={1}
```
Comment (mark-prins) [inline client/packages/system/src/Encounter/DetailView/DetailView.tsx:100]: 🤔 perhaps
```
height={200}
```
is more accurate?

**Review by mark-prins — APPROVED** [review]
👍 great solution to pick your way through so many competing requirements!
Working nicely - only extra suggestion would be some more words on the dialog.. but I've got no suggestions. perhaps Richard will suggest something later.

File: client/packages/programs/src/schema_types.ts (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/programs/src/schema_types.ts]: Sorry have to remember to check these myself!

File: client/packages/system/src/Encounter/DetailView/DetailView.tsx (hunk 2 above)
Comment (clemens-msupply) [reply client/packages/system/src/Encounter/DetailView/DetailView.tsx]: Thanks did this

**Comment by richardmoizeau** [comment]
> 👍 great solution to pick your way through so many competing requirements! Working nicely - only extra suggestion would be some more words on the dialog.. but I've got no suggestions. perhaps Richard will suggest something later.

Something like: 

by saving as "Visited", you confirm that the meeting has taken place. The status of the meeting will be changed from "Current Status" to Visited.

**Pull request merged**